README.md

```diff
@@ -102,6 +101,44 @@ int main() {
 }
 ```
 
+By design, the body of a dispatch could be any code. While member function is one useful pattern supported by macro `PRO_DEF_MEMBER_DISPATCH`, free function is also supported with another macro `PRO_DEF_FREE_DISPATCH`. The following example uses `PRO_DEF_FREE_DISPATCH` and `std::invoke` to implement similar function wrapper as `std::function` and `std::move_only_function` and supports multiple overloads.
+
+```cpp
+// Abstraction (poly is short for polymorphism)
+namespace poly {
+
+template <class... Overloads>
+PRO_DEF_FREE_DISPATCH(Call, std::invoke, Overloads...);
+template <class... Overloads>
+PRO_DEF_FACADE(MovableCallable, Call<Overloads...>);
+template <class... Overloads>
+PRO_DEF_FACADE(CopyableCallable, Call<Overloads...>, pro::copyable_pointer_constraints);
+
+}  // namespace poly
+
+// MyFunction has similar functionality as std::function but supports multiple overloads
+// MyMoveOnlyFunction has similar functionality as std::move_only_function but supports multiple overloads
+template <class... Overloads>
+using MyFunction = pro::proxy<poly::MovableCallable<Overloads...>>;
+template <class... Overloads>
+using MyMoveOnlyFunction = pro::proxy<poly::CopyableCallable<Overloads...>>;
+
+int main() {
+  auto f = [](auto&&... v) {
+    printf("f() called. Args: ");
+    ((std::cout << v << ":" << typeid(decltype(v)).name() << ", "), ...);
+    puts("");
```

Comment:
remove `puts("");`? looks like it does nothing.

Reply:
It prints an empty line. Otherwise, all the output stays on a same line.